libraries/AP_OSD/AP_OSD_Setting.cpp

```diff
@@ -48,10 +48,95 @@ const AP_Param::GroupInfo AP_OSD_Setting::var_info[] = {
     AP_GROUPEND
 };
 
+const AP_Param::GroupInfo AP_OSD_ParamSetting::var_info[] = {
+    // @Param: _EN
+    // @DisplayName: Enable
+    // @Description: Enable setting
+    // @Values: 0:Disabled,1:Enabled
+    // @User: Standard
+    AP_GROUPINFO("_EN", 1, AP_OSD_ParamSetting, enabled, 0),
+
+    // @Param: _X
+    // @DisplayName: X position
+    // @Description: Horizontal position on screen
+    // @Range: 0 29
+    // @User: Standard
+    AP_GROUPINFO("_X", 2, AP_OSD_ParamSetting, xpos, 0),
+
+    // @Param: _Y
+    // @DisplayName: Y position
+    // @Description: Vertical position on screen
+    // @Range: 0 15
+    // @User: Standard
+    AP_GROUPINFO("_Y", 3, AP_OSD_ParamSetting, ypos, 0),
+
+    // @Param: _NAME
+    // @DisplayName: Parameter index
+    // @Description: Index of the parameter to be displayed and modified
+    // @User: Standard
+    AP_GROUPINFO("_IDX", 4, AP_OSD_ParamSetting, index, 0),
+
+    AP_GROUPEND
+};
+
+extern const AP_HAL::HAL& hal;
+
 // constructor
 AP_OSD_Setting::AP_OSD_Setting(bool _enabled, uint8_t x, uint8_t y)
 {
     enabled = _enabled;
     xpos = x;
     ypos = y;
 }
+
+// constructor
+AP_OSD_ParamSetting::AP_OSD_ParamSetting(uint8_t param_number, bool _enabled, uint8_t x, uint8_t y, uint16_t idx)
+    : AP_OSD_Setting(_enabled, x, y)
+{
+    index = idx;
+    _number = param_number;
+}
+
+// constructor
+AP_OSD_ParamSetting::AP_OSD_ParamSetting(uint8_t param_number, bool _enabled, uint8_t x, uint8_t y, const char* name)
+    : AP_OSD_Setting(_enabled, x, y)
+{
+    _number = param_number;
+    index = -1;
+    strcpy(_name, name);
+}
+
+// update the contained parameter
+void AP_OSD_ParamSetting::update()
+{
+    if (_current_index == index && index >= 0) {
+        return;
+    }
+
+    AP_Param::ParamToken token;
+    // a name was specified as the default
+    if (_name[0] && !index.configured()) {
+        param = AP_Param::find(_name, &type);
+        uint16_t count = 0;
+        for (AP_Param* p=AP_Param::first(&token, &type); p && param != p; p=AP_Param::next_scalar(&token, &type)) {
+            count++;
+        }
+        index = count;
+    } else {
+        uint16_t count=0;
+        for (param=AP_Param::first(&token, &type);
+            param && count < index;
+            param=AP_Param::next_scalar(&token, &type)) {
+            count++;
+        }
+    }
+
+    _current_index = index;
+
+    if (param == nullptr) {
+        enabled = false;
+    } else {
+        param->copy_name_token(token, _name, 16);
+        _name[16] = 0;
```

Comment:
This doesn't make a lot of sense - what if _name is short?  Does `copy_name_taken` null-terminate?

Reply:
It seems to, except if its the full width. Not sure what else I can do?